src/modules/hip/kernel/normalize.hpp

```diff
@@ -0,0 +1,1925 @@
+                break;
+            }
+            paramIndex += (maxParamVolume > 1) ? (rpp_hip_mod(coord, paramShape[i]) * paramStrides[i]) : 0;
+        }
+
+        extern __shared__ float sh_mem[];
+        sh_mem[hipThreadIdx_x] = 0.0f;
+        __syncthreads();
+
+        if(isValid && id_x < maxBufferLength)
+            atomicAdd(&sh_mem[paramIndex], srcPtr[srcIdx]);
+        __syncthreads();
+
+        if (hipThreadIdx_x < maxParamVolume)
+            atomicAdd(&meanTensor[paramBase + hipThreadIdx_x], sh_mem[hipThreadIdx_x]);
+    }
+}
+
+__global__ void final_reduction_nd_hip_tensor(float *meanTensor,
+                                              float *stdDevTensor,
+                                              uint *paramShapeTensor,
+                                              uint *roiTensor,
+                                              uint numDims,
+                                              uint maxParamVolume,
+                                              bool isMean)
+{
+    uint id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint *paramShape = &paramShapeTensor[id_z * numDims];
+    uint *roi = &roiTensor[id_z * numDims * 2 + numDims];
+
+    uint divisionFactor = 1;
+    uint paramVolume = 1;
+    for(int i = 0; i < numDims; i++)
+    {
+        paramVolume *= paramShape[i];
+        if(paramShape[i] == 1)
+            divisionFactor *= roi[i];
+    }
+
+    if(id_x >= paramVolume)
+        return;
+
+    uint paramIndex = id_z * maxParamVolume + id_x;
+    if(isMean)
+        meanTensor[paramIndex] = meanTensor[paramIndex] / divisionFactor;
+    else
+        stdDevTensor[paramIndex] = sqrtf(stdDevTensor[paramIndex] / divisionFactor);
+}
+
+__global__ void compute_stddev_nd_hip_tensor(float *srcPtr,
+                                             uint *srcMaxDims,
+                                             uint *srcStrides,
+                                             float *meanTensor,
+                                             float *stdDevTensor,
+                                             uint *roiTensor,
+                                             uint *paramShapeTensor,
+                                             uint *paramStridesTensor,
+                                             uint maxParamVolume,
+                                             uint numDims,
+                                             uint maxBufferLength)
+{
+    uint id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint *roi = &roiTensor[id_z * numDims * 2 + numDims];
+    uint *paramShape = &paramShapeTensor[id_z * numDims];
+    uint *paramStrides = &paramStridesTensor[id_z * numDims];
+    uint srcIdx = id_z * maxBufferLength + id_x;
+    uint paramBase = id_z * maxParamVolume;
+    uint paramIndex = 0;
+
+    if(maxParamVolume > MAX_SHARED_MEMORY_SIZE)
```

Comment:
When do we encounter this?

Reply:
The values of maxParamVolume depends on input tensor and axisMask value
Let's say if input size is 1920x1080 and axisMask = 1
Then paramVolume = 1080

I have defined MAX_SHARED_MEMORY_SIZE as 1024. So in this cases the value exceeds and enters the unoptimized generic version